guides/tutorials/codelabs/managed-pipeline-templates/index.md

```diff
@@ -0,0 +1,245 @@
+    instanceType: "{{ config.instanceType }}"
+    interestingHealthProviderNames:
+    - Amazon
+    keyPair: "{{ config.keyPair|default('acme-master-keypair') }}"
+    provider: aws
+    securityGroups: []
+    stack: "{{ stack }}"
+    strategy: redblack
+    maxRemainingAsgs: 2
+    useSourceCapacity: true
+    scaleDown: true
+    capacity:
+      desired: "{{ config.desiredCapacity|default(1) }}"
+      max: "{{ config.maxCapacity|default(1) }}"
+      min: "{{ config.minCapacity|default(1) }}"
+    loadBalancers:
+    - "{{ application }}-frontend"
+    subnetType: acmeInternal
+    suspendProcesses: []
+    tags: {}
+    targetHealthyDeployPercentage: 100
+    terminationPolicies:
+    - Default
+    useAmiBlockDeviceMappings: false
+```
+{% endraw %}
+
+It's a lot of configuration: It's actually a close representation to the data
+stored in a normal pipeline, but expressed as YAML instead of JSON. Let's review
+the note tags:
+
+1. The template `id` is a (globally unique) way to reference a template while
+   creating a pipeline. We'll later create a pipeline that references this.
+2. The Spinnaker UI supports creating a Pipeline from a Template, which will
+   pull this information to give better context about its purpose. The `scopes`
+   list is used to determine what apps can see the template. Global will be
+   available to any app, but if you defined `foo`, it'd be explicitly available
+   to the `foo` Spinnaker application.
+3. Template variables can be used throughout a template (or its children) and
+   support a variety of types, an object (map) is especially useful while
+   invoking modules, but isn't self-documenting. It's best practice to provide
+   an example so you can lead usage of the variables correctly.
+4. Template stages are defined as a list, but can produce any stage graph that
+   you normally could through the UI. Stages can fork, join and even inject
+   themselves into parent template graphs.
+5. Many parts of the Pipeline Template syntax supports Jinja, in this specific
+   case, we're looping over a list of regions to determine what AWS regions to
+   bake AMIs in.
+6. Another example of Jinja, one that is invoking a module, a reusable block of
+   configuration, with the object variable defined in `(3)`. You'll notice in
+
+# Publishing into Spinnaker
+
+Once a template has been written, it needs to be published somewhere before it
+can be used. Spinnaker supports a few different options out of the box for
+Template Loaders:
+
+* `file`: Load a pipeline template off the machine running Orca.
+* `http`: Load a pipeline template from an unauthenticated HTTP resource.
+* `spinnaker`: Load a pipeline template from Spinnaker itself.
+
+In this section, we'll focus on the recommended loader, `spinnaker`, as it
+provides the greatest gamut of features and flexibility. To publish into 
+Spinnaker, we'll use [Roer][roer-repo], a CLI app for Spinnaker that has
+commands for interacting with Pipeline Templates.
```

Comment:
The goal of roer is strictly to give a thin wrapper around the API to make getting pipeline templates and so-on into Spinnaker easier, whereas Halyard's scope is dramatically larger in that it's focused on configuration & operations of Spinnaker itself. If someone is just looking to have a client to publish templates from a CI job, having a zero-dependency tool has a lot of advantages in my mind.